TokboxScreenPublisher.js

```diff
@@ -0,0 +1,155 @@
+      }
+    },
+    mediaStopped: () => {
+      notification.warning({
+        duration: notificationDuration,
+        message: 'Screenshare stream stopped',
+        description: 'If this is not intentional check that the shared window has not been closed.'
+      })
+    }
+  })
+}
+
+function onScreenPublisherError (error) {
+  const { name, code, message } = error
+  log('Screenshare publish error', error)
+  
+  if (!screenPublisher) {
+    emitLocalEvent('ScreenSharing:disabled')
+    emitLocalEvent('TokboxSession:initPublisher:error', { error, videoType: 'screen' })
+  }
+  
+  if (!creatingScreenPublisher) return
+  
+  creatingScreenPublisher.destroy()
+  creatingScreenPublisher = null
+
+  if (name !== 'OT_USER_MEDIA_ACCESS_DENIED') {
+    notification.warning({
+      duration: notificationDuration,
+      message: 'Screenshare stream failed',
+      description: 'Establishing screenshare stream failed. Please try again.'
+    })
+    Raven.captureException(error)
+  }
+}
+
+function createScreenPublisher ({ name, lastname }) {
+  let myPublisher = OpenTok.initPublisher({
+    ...publisherOptions,
+    frameRate: 30, // Test 15
+    videoSource: 'screen',
+    publishVideo: true,
+    name: name + (lastname ? ' ' + lastname : '') + ' shared screen'
+  }, (error) => error && onScreenPublisherError(error))
+
+  setScreenPublisherHandlers(myPublisher)
+
+  return myPublisher
+}
+
+function tokboxScreenPublish ({ screenSharedEnabled, screenSharedAnother, name, lastname }) {
+  const thePublisher = creatingScreenPublisher || screenPublisher
+
+  if (thePublisher && !screenSharedEnabled) {
+    if (creatingScreenPublisher) creatingScreenPublisher.destroy()
+    if (screenPublisher) screenPublisher.destroy()
+    creatingScreenPublisher = null
+    screenPublisher = null
+
+    return log('Screenshare has been disabled')
+  }
+
+  if (!screenSharedEnabled 
+    || creatingScreenPublisher
+    || (screenPublisher && !screenSharedAnother)) {
+    
+    return log('Screenshare is not been enabled or is already enabled')
+  }
+
+  creatingScreenPublisher = createScreenPublisher({ name, lastname })
+  if (screenSharedAnother) emitLocalEvent('TokboxSession:screenSharedAnother')
+
+  if (!creatingScreenPublisher) {
+    return log('Screenshare publisher not created')
+  }
+
+  return tokboxSession.publish(creatingScreenPublisher, (error) => {
+    if (error) {
+      return onScreenPublisherError(error)
+    }
+    if (screenPublisher) {
+      screenPublisher.demioScreenSharedAnother = screenSharedAnother
+      screenPublisher.destroy()
+      log('Screenshare stream successfully replaced by another')
+    }
+    screenPublisher = creatingScreenPublisher
+    creatingScreenPublisher = null
+  })
+}
+
+function tokboxScreenUnPublish () {
+  const [ screenSharedStream ] = tokboxSession.streams.where({ videoType: 'screen' })
+  if (screenSharedStream) tokboxSession.forceUnpublish(screenSharedStream, (error) => {
```

Comment:
Why forceUnpublish? Are you just unpublishing your own screen right? I would use [unpublish](https://tokbox.com/developer/sdks/js/reference/Session.html#unpublish).

`forceUnpublish` is usually used by user with role "moderator" to unpublish other participants streams

Reply:
We allow one screen sharing only. When a presenter start sharing we force unpublish the previous screen sharing of the other presenter. They act like moderators.

Reply:
Got it 👍